crates/re_space_view_spatial/src/spatial_topology.rs

```diff
@@ -0,0 +1,698 @@
+                new_space.entities.insert(e.clone());
+                false
+            } else {
+                true
+            }
+        });
+
+        // Transfer any child spaces from self to the new space if they're children of the new space.
+        split_subspace
+            .child_spaces
+            .retain(|child_origin, connections| {
+                debug_assert!(child_origin != new_space_origin);
+
+                if child_origin.is_descendant_of(new_space_origin) {
+                    new_space
+                        .child_spaces
+                        .insert(child_origin.clone(), *connections);
+                    false
+                } else {
+                    true
+                }
+            });
+
+        // Note that the new connection information may change the known dimensionality of the space that we're splitting.
+        split_subspace.add_or_update_child_connection(new_space_origin, connections);
+
+        // Patch parents of the child spaces that were moved to the new space.
+        for child_origin in new_space.child_spaces.keys() {
+            let child_space = self
+                .subspaces
+                .get_mut(&child_origin.hash())
+                .expect("Child origin not part of origin->subspace map.");
+            child_space.parent_space = Some(new_space.origin.hash());
+        }
+
+        self.subspaces.insert(new_space.origin.hash(), new_space);
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use re_log_types::EntityPath;
+    use re_types::{
+        components::{DisconnectedSpace, PinholeProjection},
+        ComponentName, Loggable as _,
+    };
+
+    use crate::spatial_topology::{SubSpaceConnectionFlags, SubSpaceDimensionality};
+
+    use super::SpatialTopology;
+
+    #[test]
+    fn no_splits() {
+        let mut topo = SpatialTopology::default();
+
+        // Initialized with root space.
+        assert_eq!(topo.subspaces.len(), 1);
+        assert_eq!(topo.subspace_origin_per_logged_entity.len(), 0);
+
+        // Add a simple tree without any splits for now.
+        add_diff(&mut topo, "robo", &[]);
+        add_diff(&mut topo, "robo/arm", &[]);
+        add_diff(&mut topo, "robo/eyes/cam", &[]);
+
+        // Check that all entities are in the same space.
+        check_paths_in_space(&topo, &["robo", "robo/arm", "robo/eyes/cam"], "/");
+
+        // .. and that space has no children and no parent.
+        let subspace = topo.subspace_for_entity(&"robo".into());
+        assert!(subspace.child_spaces.is_empty());
+        assert!(subspace.parent_space.is_none());
+
+        // If we make up entities that weren't logged we get the closest space
+        assert_eq!(
+            topo.subspace_for_entity(&EntityPath::root()).origin,
+            EntityPath::root()
+        );
+        assert_eq!(
+            topo.subspace_for_entity(&"robo/eyes".into()).origin,
+            EntityPath::root()
+        );
+    }
```

Comment:
Since eyes is still a prefix of robo/eyes/cam, worth also checking a non-logged entity such as:
```suggestion
        assert_eq!(
            topo.subspace_for_entity(&"robo/leg".into()).origin,
            EntityPath::root()
        );
    }
```